WalletWasabi.Daemon/Rpc/WasabiJsonRpcService.cs

```diff
@@ -263,6 +264,74 @@ public string PayInCoinJoin(BitcoinAddress address, Money amount)
 		return activeWallet.AddCoinJoinPayment(address, amount);
 	}
 
+	[JsonRpcMethod("listpaymentsincoinjoin")]
+	public JsonRpcResultList ListPaymentsInCoinJoin()
+	{
+		var activeWallet = Guard.NotNull(nameof(ActiveWallet), ActiveWallet);
+		AssertWalletIsLoaded();
+		var payments = activeWallet.BatchedPayments.GetPayments();
+		return payments.Select(x =>
+		{
+			var paymentResult = new JsonRpcResult
+			{
+				["id"] = x.Id,
+				["amount"] = x.Amount.Satoshi,
+				["destination"] = x.Destination.ScriptPubKey.ToHex()
+			};
+
+			var state = x.State;
+			var stateHistory = new List<JsonRpcResult>();
+			while (state != null)
+			{
+				switch (state)
+				{
+					case PendingPayment pending:
+						stateHistory.Add(new JsonRpcResult
+						{
+							["status"] = "Pending"
+						});
+						break;
+
+					case InProgressPayment inProgress:
+						stateHistory.Add(new JsonRpcResult
+						{
+							["status"] = "In progress",
+							["round"] = inProgress.RoundId.ToString()
+						});
+						break;
+
+					case FinishedPayment finished:
+						stateHistory.Add(new JsonRpcResult
+						{
+							["status"] = "Finished",
+							["txid"] = finished.TransactionId.ToString()
+						});
+						break;
+
+					default:
+						throw new NotSupportedException($"Unrecognized state: {state.GetType().Name}.");
+				}
+
+				state = state.PreviousState;
```

Comment:
In which cases this line will be executed ?

Reply:
It will be executed always.